xapian-core/tests/api_closedb.cc

```diff
@@ -242,6 +299,20 @@ DEFINE_TESTCASE(closedb5, transactions) {
 	Xapian::Database db = get_writable_database_as_database();
 	TEST_EQUAL(db.get_doccount(), 0);
     }
+
+    {
+	// commit_transaction() throws InvalidOperationError when
+	// not in a transaction. begin_transaction() is no-op or
+	// throws DatabaseError
+	Xapian::WritableDatabase wdb = get_writable_database();
+	wdb.close();
+	TEST_EXCEPTION(Xapian::InvalidOperationError,
+		wdb.commit_transaction());
+	try {
+	    wdb.begin_transaction();
```

Comment:
If `begin_transaction()` succeeds it would be good to test that `commit_transaction()` does something appropriate.

Reply:
Checked begin_transaction() returns successfully, based on that commit_transaction() check is done.  Let me know if it needs any update.